src/uu/pgrep/src/pgrep.rs

```diff
@@ -0,0 +1,188 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+pub mod pid;
+
+use std::{borrow::BorrowMut, cmp::Ordering};
+
+use clap::{arg, crate_version, Arg, ArgAction, ArgGroup, Command};
+use pid::walk_pid;
+use uucore::{error::UResult, format_usage, help_about, help_usage};
+
+const ABOUT: &str = help_about!("pgrep.md");
+const USAGE: &str = help_usage!("pgrep.md");
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+
+    let default_delimiter = "\n".into();
+    let delimiter = matches
+        .get_one::<String>("delimiter")
+        .unwrap_or(&default_delimiter);
+
+    let mut programs = matches
+        .get_many::<String>("pattern")
+        .unwrap_or_default()
+        .map(|it| it.to_string())
+        .collect::<Vec<_>>();
```

Comment:
I think this is a bit misleading because there can be only one pattern (or none).

Reply:
can you give examples of this usage?

Reply:
An example:
```
$ pgrep -o
1
$ pgrep -o konsole
1245
$ pgrep -o konsole firefox
pgrep: only one pattern can be provided
```

Reply:
fixed 

```shell
❯ cargo run -p uu_pgrep -- -o zsh zsh
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.03s
     Running `/home/krysztal/Desktop/project/uutils/procps/target/debug/pgrep -o zsh zsh`
pgrep: only one pattern can be provided
```